batch/batch/worker/worker.py

```diff
@@ -343,6 +344,29 @@ async def wrapper(f, *args, **kwargs):
     return wrapper
 
 
+class TaskCancelledError(Exception):
+    pass
+
+
+async def run_until_done_or_cancelled(event: asyncio.Event, f: Callable[[...], Awaitable[Any]], *args, **kwargs):
+    task = asyncio.ensure_future(f(*args, **kwargs))
+    cancelled = asyncio.ensure_future(event.wait())
+    try:
+        await asyncio.wait([cancelled, task], return_when=asyncio.FIRST_COMPLETED)
+        if cancelled.done():
+            raise TaskCancelledError
```

Comment:
I fear this naming convention is going to make communication more confusing. "Task" and "cancel" are both asyncio concepts, and while this is similar in intention to an `asyncio.CancelledError`, it is not, and it operates on a different level of abstraction (our application code vs library code). I also really hesitate to call anything in our system a Task, to prevent confusion with asyncio Tasks and the discrepancy with the usual use of Task in the literature (from the papers that we've read so far in systems cs I believe Task is generally equivalent to our notion of a Job).

Reply:
If you want to generalize from deleted, I think stopped is better, but it might be best not to do the abstraction until it is necessary.